src/main/java/edu/tamu/sage/job/JobRunner.java

```diff
@@ -0,0 +1,98 @@
+package edu.tamu.sage.job;
+
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Service;
+
+import edu.tamu.sage.model.Job;
+import edu.tamu.sage.model.repo.JobRepo;
+import edu.tamu.sage.service.ProcessorService;
+
+@Service
+public class JobRunner {
+    @Autowired
+    JobRepo jobRepo;
+
+    @Autowired
+    ProcessorService processorService;
+
+    private Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private final DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US);
+
+    private static final String START_TIME_KEY = "startTime";
+    private static final String DATE_KEY = "date";
+    private static final String DAYS_KEY = "days";
+    private static final String MONTHS_KEY = "months";
+
+    @Scheduled(cron = "0 0/30 * * * ?")
+    private void runJobs() {
+        logger.debug("Checking "+jobRepo.findByScheduleActiveTrue().size()+" Active Jobs");
+        LocalDateTime schedulerStarted = java.time.LocalDateTime.now(ZoneId.of("UTC"));
+        jobRepo.findByScheduleActiveTrue().forEach(j -> {
+            switch (j.getSchedule().getFrequency()) {
```

Comment:
The following is not required for approval. An alternative approach: store enum frequency, ONCE and a subset of ChronoUnit, DateTime startTime, and boolean active on schedule. For @Scheduled runJob, query for active true and past startTime. Run jobs returned and if frequency ONCE, set active to false and save, else update startTime by casting frequency to ChronoUnit and add to startTime and save. This will simplify both backend logic and front end form significantly. Might be ideal to add modTime to schedule with @UpdateTimestamp to know when it was last updated either by schedule or user change.